bugs_s2.py

```
@@ -0,0 +1,787 @@
+    train_primary_with_ensemble(primary_model, auxiliary_model, optimizer, data, target)
+
+
+def resilient_training_step(model, data, target, optimizer):
+    """Training step with error resilience"""
+    optimizer.zero_grad()
+
+    try:
+        output = model(data)
+
+        if torch.rand(1).item() > 0.5:
+            raise ValueError("Unstable training condition detected")
```

Ruff (0.13.2)

Abstract `raise` to an inner function

(TRY301)